src/error/decode.rs

```diff
@@ -341,9 +341,12 @@ impl DecodeError {
     fn from_codec_kind(inner: CodecDecodeError) -> Self {
         let codec = match inner {
             CodecDecodeError::Ber(_) => crate::Codec::Ber,
+            #[allow(unreachable_patterns)]
```

Comment:
How are these unreachable?

Reply:
These types are currently empty so they never have value, for example

```rust
#[derive(Snafu, Debug)]
#[snafu(visibility(pub))]
#[non_exhaustive]
pub enum CerDecodeErrorKind {}
```